### ISSUE TITLE
slurm --cluster support

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc#2504]
The latest release of slurm apparently supports a single unique job ID across a federated cluster.  For the moment though, if you submit a job to slurm on host X with `#SBATCH --cluster = Y` to make it run on host Y, any subsequent job interaction via the resulting job ID has to be done on host Y or else with `--cluster=Y` on the command line (i.e. the job ID is not recognized on the original submission host).  

This way of submitting remote jobs without ssh is fine with Cylc, if hosts X and Y see the same filesystem (i.e. the job looks local to Cylc, even though it technically isn't).  But with slurm subsequent job poll or kill fails because the Job ID is not recognized locally, and Cylc does not know to use `--cluster=Y` on the  squeue and scancel command lines.

We should either make this work in Cylc or else document the problem and recommend using remote mode for the moment.

**Comment by hjoliver** [comment]
It is easy to make a custom batch system handler for a hardwired specific cluster. E.g. for cluster foo, make `slurm_cluster_foo.py`:

```python
from cylc.batch_sys_handlers.slurm import SLURMHandler

class SLURMClusterKupeHandler(SLURMHandler):
    """SLURM job submission and manipulation for --cluster=foo."""
    KILL_CMD_TMPL = "scancel --cluster=foo '%(job_id)s'"
    POLL_CMD = "squeue -h --cluster=foo"
    SUBMIT_CMD_TMPL = "sbatch --cluster=foo '%(job)s'"

BATCH_SYS_HANDLER = SLURMClusterKupeHandler()
```

Obviously it would be better not to have a single slurm handler that extracts the cluster name from job directives though.  Different jobs could potentially use different clusters.

Without altering the core of Cylc, we could have the job submit, poll, and kill commands all extract `--cluster=foo` from the job script before executing - every time.  This seems a bit perverse though.

Another option would be to have the task proxy remember the cluster name, if provided, and pass it to the job submit, poll, and kill methods each time.   At a cost of one new task proxy attribute that will only every be used by slurm jobs.

@cylc/core - any strong opinions on this?  Or other ideas?

**Comment by arjclark** [comment]
@hjoliver - Thinking about this the other way round, could we maybe provide a config option under [job] that would be inserted into slurm directives (ignored otherwise) and used in the commands accordingly? (similar to the execution time limit entry)

**Comment by matthewrmshin** [comment]
We are likely to make life a lot easier for this issue when we solve #2199. (We'll schedule the work to commence after #2468 is merged.)

**Comment by hjoliver** [comment]
Trouble is we need this already on the new HPC at NIWA (although we could make do with the nasty hard-wired kludge above)

**Comment by matthewrmshin** [comment]
@hjoliver Understood. I think it is best to use a custom batch system handler for now. I have really wanted to work on #2199 before the end of this year, but that hasn't happened.

**Comment by hjoliver** [comment]
From local HPC engineer:
> Federation doesn’t look an appropriate way forward:
>
> *“A job is submitted to the local cluster (the cluster defined in the slurm.conf) and is then replicated across the clusters in the federation. Each cluster then independently attempts to the schedule the job based off of its own scheduling policies.“*
